internal/builtins/rest.ts

```diff
@@ -22,21 +22,26 @@ export type RequestOutput = {
   response: Record<string, unknown> | string;
 };
 
+export type InputOptions = {
+  headers?: Record<string, unknown>;
```

Comment:
If we accept `null` in some places for optional fields, let's accept it everywhere.

```suggestion
  headers?: Record<string, unknown> | null;
```

Reply:
Headers and urlParams need to be `{}` and not null or undefined :/ - so they're not actually optional but we do have a default value.

```
let test = {"a" : null}
let {a = "what"} = test // a is null and not what
let {none = "what2"} = test // none is what2
```
I think letting headers be null means we have to add an extra check for if it's null vs. only handling if it's undefined or not.

Reply:
I'm referring to what we accept as input, which can differ from what we pass to the API. Can we just convert both `null` and `undefined` to `{}`? `headers = headers ?? {}`